spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -111,7 +110,7 @@ public Object execute(Object[] parameters) {
 		Collector<?, ?, ?> collector;
 
 		if (this.tree.isCountProjection()) {
-			collector = Collectors.reducing(0, e -> 1, Integer::sum);
+			collector = Collectors.counting();
```

Comment:
These changes are going to clash with the other PR. @ChengyuanZhao

Reply:
I think it should be OK if either one goes in first. It will take some manual merging but I don't think anything is in contradictory.